File: test-chatanywhere.js
/**
 * Test script for ChatAnywhere API integration
 * Run with: node test-chatanywhere.js
 */

const axios = require('axios');

// API Configuration
const API_KEY = 'sk-POcyyRhXrzVwwPedbzrHqfQgNNqslFSXTcgR3KEakZpdzzte';
const API_BASE = 'https://api.chatanywhere.tech/v1';

// Available models
const MODELS = {
  'claude-sonnet': 'claude-sonnet-4-5-20250929',
  'claude-thinking': 'claude-sonnet-4-5-20250929-thinking',
  'gpt-5': 'gpt-5'
};

// Test function for each model
async function testModel(modelKey, modelName) {
  console.log(`\n🔄 Testing ${modelKey} (${modelName})...`);
  
  try {
    const response = await axios.post(
      `${API_BASE}/chat/completions`,
      {
        model: modelName,
        messages: [
          {
            role: 'system',
            content: 'You are a helpful assistant. Keep your response brief.'
          },
          {
            role: 'user',
            content: 'Say hello and tell me which AI model you are in one sentence.'
          }
        ],
        temperature: 0.7,
        max_tokens: 100
      },
      {
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${API_KEY}`
        },
        timeout: 30000 // 30 seconds timeout
      }
    );

    if (response.data && response.data.choices && response.data.choices.length > 0) {
      console.log(`✅ ${modelKey} Success!`);
      console.log(`   Response: ${response.data.choices[0].message.content}`);
      console.log(`   Usage: ${JSON.stringify(response.data.usage)}`);
      return true;
    } else {
      console.log(`⚠️ ${modelKey} returned unexpected response structure`);
      return false;
    }
  } catch (error) {
    console.log(`❌ ${modelKey} Failed!`);
    if (error.response) {
      console.log(`   Status: ${error.response.status}`);
      console.log(`   Error: ${error.response.data?.error?.message || error.message}`);
    } else if (error.request) {
      console.log(`   Error: No response received (network issue)`);
    } else {
      console.log(`   Error: ${error.message}`);
    }
    return false;
  }
}

// Main test function
async function runTests() {
  console.log('🚀 Starting ChatAnywhere API Tests');
  console.log('================================');
  console.log(`API Base: ${API_BASE}`);
  console.log(`API Key: ${API_KEY.substring(0, 10)}...${API_KEY.substring(API_KEY.length - 5)}`);
  
  const results = [];
  
  // Test each model
  for (const [key, model] of Object.entries(MODELS)) {
    const success = await testModel(key, model);
    results.push({ key, model, success });
    
    // Add delay between requests to avoid rate limiting
    await new Promise(resolve => setTimeout(resolve, 1000));
  }
  
  // Summary
  console.log('\n================================');
  console.log('📊 Test Results Summary:');
  console.log('================================');
  
  let successCount = 0;
  results.forEach(result => {
    const status = result.success ? '✅' : '❌';
    console.log(`${status} ${result.key}: ${result.success ? 'PASSED' : 'FAILED'}`);
    if (result.success) successCount++;
  });
  
  console.log('\n================================');
  console.log(`Total: ${successCount}/${results.length} models working`);
  
  if (successCount === results.length) {
    console.log('🎉 All models are working correctly!');
  } else if (successCount > 0) {
    console.log('⚠️ Some models are working, but not all. Check the errors above.');
  } else {
    console.log('❌ No models are working. Please check your API key and network connection.');
  }
}

// Run tests
runTests().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});

File: test_chatanywhere.py
#!/usr/bin/env python3
"""
Test script for ChatAnywhere API integration with Python
Run with: python test_chatanywhere.py
"""

import os
import json
import time
from typing import Dict, List, Tuple

# Try to import openai, provide instructions if not installed
try:
    import openai
except ImportError:
    print("❌ OpenAI package not installed!")
    print("Please install it with: pip install openai")
    print("Or: python -m pip install openai")
    exit(1)

# API Configuration
API_KEY = "sk-POcyyRhXrzVwwPedbzrHqfQgNNqslFSXTcgR3KEakZpdzzte"
API_BASE = "https://api.chatanywhere.tech/v1"

# Available models
MODELS = {
    'claude-sonnet': 'claude-sonnet-4-5-20250929',
    'claude-thinking': 'claude-sonnet-4-5-20250929-thinking',
    'gpt-5': 'gpt-5'
}

def test_model(client: openai.OpenAI, model_key: str, model_name: str) -> Tuple[bool, str]:
    """
    Test a specific model
    
    Args:
        client: OpenAI client instance
        model_key: Short key for the model
        model_name: Full model name
        
    Returns:
        Tuple of (success: bool, message: str)
    """
    print(f"\n🔄 Testing {model_key} ({model_name})...")
    
    try:
        response = client.chat.completions.create(
            model=model_name,
            messages=[
                {"role": "system", "content": "You are a helpful assistant. Keep your response brief."},
                {"role": "user", "content": "Say hello and tell me which AI model you are in one sentence."}
            ],
            temperature=0.7,
            max_tokens=100
        )
        
        if response and response.choices:
            content = response.choices[0].message.content
            print(f"✅ {model_key} Success!")
            print(f"   Response: {content}")
            if hasattr(response, 'usage'):
                print(f"   Tokens used: {response.usage.total_tokens}")
            return True, content
        else:
            print(f"⚠️ {model_key} returned unexpected response")
            return False, "Unexpected response structure"
            
    except Exception as e:
        print(f"❌ {model_key} Failed!")
        print(f"   Error: {str(e)}")
        return False, str(e)

def test_all_models() -> Dict[str, bool]:
    """
    Test all available models
    
    Returns:
        Dictionary with model results
    """
    print('🚀 Starting ChatAnywhere API Tests')
    print('=' * 40)
    print(f'API Base: {API_BASE}')
    print(f'API Key: {API_KEY[:10]}...{API_KEY[-5:]}')
    
    # Initialize client
    client = openai.OpenAI(
        api_key=API_KEY,
        base_url=API_BASE
    )
    
    results = {}
    
    # Test each model
    for key, model in MODELS.items():
        success, message = test_model(client, key, model)
        results[key] = success
        
        # Add delay to avoid rate limiting
        time.sleep(1)
    
    return results

def print_summary(results: Dict[str, bool]) -> None:
    """
    Print test results summary
    
    Args:
        results: Dictionary with test results
    """
    print('\n' + '=' * 40)
    print('📊 Test Results Summary:')
    print('=' * 40)
    
    success_count = 0
    for key, success in results.items():
        status = '✅' if success else '❌'
        result = 'PASSED' if success else 'FAILED'
        print(f'{status} {key}: {result}')
        if success:
            success_count += 1
    
    print('\n' + '=' * 40)
    print(f'Total: {success_count}/{len(results)} models working')
    
    if success_count == len(results):
        print('🎉 All models are working correctly!')
    elif success_count > 0:
        print('⚠️ Some models are working, but not all.')
    else:
        print('❌ No models are working. Check your API key.')

def example_usage():
    """
    Show example usage of the ChatAnywhere API
    """
    print('\n' + '=' * 40)
    print('📚 Example Usage in Your Code:')
    print('=' * 40)
    
    example_code = '''
# Initialize the client
import openai

client = openai.OpenAI(
    api_key="sk-POcyyRhXrzVwwPedbzrHqfQgNNqslFSXTcgR3KEakZpdzzte",
    base_url="https://api.chatanywhere.tech/v1"
)

# Use Claude Sonnet
response = client.chat.completions.create(
    model="claude-sonnet-4-5-20250929",
    messages=[
        {"role": "user", "content": "Hello!"}
    ]
)
print(response.choices[0].message.content)

# Use GPT-5
response = client.chat.completions.create(
    model="gpt-5",
    messages=[
        {"role": "user", "content": "Explain quantum computing"}
    ]
)
print(response.choices[0].message.content)
'''
    print(example_code)

def main():
    """
    Main function
    """
    try:
        # Run tests
        results = test_all_models()
        
        # Print summary
        print_summary(results)
        
        # Show example usage
        example_usage()
        
    except KeyboardInterrupt:
        print('\n\n⚠️ Tests interrupted by user')
    except Exception as e:
        print(f'\n❌ Fatal error: {e}')
        exit(1)

if __name__ == '__main__':
    main()
